sqlserver/datadog_checks/sqlserver/statements.py

```diff
@@ -350,14 +351,16 @@ def _to_metrics_payload_row(row):
             del row['procedure_text']
         return row
 
-    def _to_metrics_payload(self, rows):
+    def _to_metrics_payload(self, rows, max_queries):
+        # sort by total_elapsed_time and return the top max_queries
+        rows.sort(key=lambda i: i['total_elapsed_time'], reverse=True)
```

Comment:
This operation will have side effects. It sorts the passed list in-place. 

Might be better to move the truncation of max query metrics to this line so all the logic for limiting the max queries is in one spot too.

```python
rows = rows.sorted(key=lambda i: i['total_elapsed_time'], reverse=True)
```